src/sql/src/plan/transform_ast.rs

```diff
@@ -199,6 +223,61 @@ impl<'a> FuncRewriter<'a> {
         Self::plan_variance(expr, filter, distinct, sample).call_unary(vec!["sqrt"])
     }
 
+    fn plan_bool_and(
+        &self,
+        expr: Expr<Aug>,
+        filter: Option<Box<Expr<Aug>>>,
+        distinct: bool,
+    ) -> Expr<Aug> {
+        // The code below converts `bool_and(x)` into:
+        //
+        //     sum((NOT x)::int4) = 0
+        //
+        // It is tempting to use `count` instead, but count does not return NULL
+        // on empty or all NULL inputs, as required.
```

Comment:
I think this should be "*any* NULL inputs".

Reply:
Wait, but `count` doesn't return `NULL` on any `NULL` input. It just returns `NULL` when all inputs are `NULL`? (Including the case of zero input values.)

Reply:
(We talked in person, and I believe this is correct as written, and SQL is just massively confusing wrt its semantics here.)